packages/react/src/publication/adapters/ReactionGateway.ts

```diff
@@ -12,10 +12,27 @@ import {
 import { ReactionRequest, IReactionGateway } from '@lens-protocol/domain/use-cases/publications';
 import { assertError } from '@lens-protocol/shared-kernel';
 
-export class ReactionGateway implements IReactionGateway {
+export type ExtendedReactionRequest = ReactionRequest & {
+  existingReactionType?: ReactionRequest['reactionType'];
```

Comment:
would avoid using `?` as it's easy to miss that, if `undefined` is correct value let's state it clearly in union `ReactionRequest['reactionType'] | undefined`